### DISCUSSION TITLE
i need custom menus

### DISCUSSION THREAD

**Discussion by BernardoGP4504** [YoshiCrafter29/YoshiCrafterEngine#195] [Q&A]
i searched EVERYWHERE and didn't found how to make custom menus but have seen people/articles mentioning it
Does it needs to be with `.hx` files?

**Comment by IMdaAG** [comment]
Yes it needs to be an `.hx` file. Then the file(s) you put in `YOUR MOD/ui`. If you don't have the `ui` folder just create it. And btw you need to name it the same as the hx files from source (For example you want to edit the main menu the name of the `.hx` file would be `MainMenuState.hx`)

**Reply by BernardoGP4504** [reply]
sure but there's an error, i copied and pasted the same file from this github and in the import things there where a lot of errors i commented out those lines but now everytime i open my mod it just gives an error:
```
Failed to parse the file located at "./mods/Hex mod port/ui/MainMenuState.hx".
hscript:384: Invalid preprocessor (Unclosed) at 384
```
but that doesn't crash the engine
